packages/server/lib/automation/cookies.ts

```diff
@@ -31,9 +32,20 @@ const normalizeCookieProps = function (props) {
     return props
   }
 
+  // if the cookie is stored inside the server side cookie jar,
+  // we want to make the automation client aware so the domain property
+  // isn't mutated to prevent duplicate setting of cookies from different contexts.
+  // This should be handled by the hostOnly property
+
   const cookie = _.pick(props, COOKIE_PROPERTIES)
 
-  if (props.expiry != null) {
+  if (props.expiry === '-Infinity') {
+    cookie.expiry = -Infinity
+    // set the cookie to expired so when set, the cookie is removed
+    cookie.expirationDate = 0
```

Comment:
Do we need to add expirationDate to the AutomationCookie types?

Reply:
wait -- is this the Chrome Cookie types? Not clear which types cookie is here

Reply:
It should be `AutomationCookie` but it isn't typed, just parsed. I can see if I can improve this to add some better typing support, which I think would help with readability. We have a lot of cookie types 😅

Reply:
I have a follow up to address some of the typings here in the automation API to make this a bit more digestible https://github.com/cypress-io/cypress/compare/fix-duplicate-and-expired-cookies...improve-automation-types?expand=1